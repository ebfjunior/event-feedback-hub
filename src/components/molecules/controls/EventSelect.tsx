import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/components/ui/select';
import type { EventListItem } from '@/types/api';

export type EventSelectProps = {
  events: EventListItem[];
  value?: string;
  onChange?: (eventId: string | undefined) => void;
};

export function EventSelect({ events, value, onChange }: EventSelectProps) {
  return (
    <Select value={value ?? 'all'} onValueChange={(v) => onChange?.(v === 'all' ? undefined : v)}>
      <SelectTrigger className="w-48">
        <SelectValue placeholder="All Events" />
      </SelectTrigger>
      <SelectContent>
        <SelectItem value="all">All Events</SelectItem>
        {events.map((e) => (
          <SelectItem key={e.id} value={e.id}>
            {e.name}
          </SelectItem>
        ))}
      </SelectContent>
    </Select>
  );
}


