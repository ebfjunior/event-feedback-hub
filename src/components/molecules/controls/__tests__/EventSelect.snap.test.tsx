import React from 'react';
import { render } from '@testing-library/react';
import { EventSelect } from '../EventSelect';

// Mock Radix Select to stable DOM
vi.mock('@/components/ui/select', () => ({
	Select: (p: any) => <div data-mock="Select" {...p} />,
	SelectContent: (p: any) => <div data-mock="SelectContent" {...p} />,
	SelectItem: (p: any) => <div data-mock="SelectItem" {...p} />,
	SelectTrigger: (p: any) => <div data-mock="SelectTrigger" {...p} />,
	SelectValue: (p: any) => <div data-mock="SelectValue" {...p} />,
}));

describe('EventSelect (snapshot)', () => {
	it('renders with events', () => {
		const { container } = render(
			<EventSelect events={[{ id: 'e1', name: 'Event 1' }, { id: 'e2', name: 'Event 2' }]} />,
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});


