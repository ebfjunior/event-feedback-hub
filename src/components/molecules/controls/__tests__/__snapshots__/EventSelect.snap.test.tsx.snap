// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EventSelect (snapshot) > renders with events 1`] = `
<div
  data-mock="Select"
  value="all"
>
  <div
    class="w-48"
    data-mock="SelectTrigger"
  >
    <div
      data-mock="SelectValue"
      placeholder="All Events"
    />
  </div>
  <div
    data-mock="SelectContent"
  >
    <div
      data-mock="SelectItem"
      value="all"
    >
      All Events
    </div>
    <div
      data-mock="SelectItem"
      value="e1"
    >
      Event 1
    </div>
    <div
      data-mock="SelectItem"
      value="e2"
    >
      Event 2
    </div>
  </div>
</div>
`;
